### PR TITLE
Adding Service Event Rules and Rules Bug Fixes

### PR DESCRIPTION
**Pull request by stmcallister**
This PR adds the `pagerduty_service_event_rule` resource, addressing #279.

It also adds `suspend` and `variables` fields to `pagerduty_ruleset_rule` mentioned in #272.

Also allows `suppress.value` to be set to false. #295, #245, #225.

Fixed bug in `pagerduty_ruleset_rule` where position wasn't setting properly. The `position` should now set correctly on the first apply for both `pagerduty_ruleset_rule` and `pagerduty_service_event_rule`.

Test results  for the `pagerduty_service_event_rule` resource are here: 

```
TF_ACC=1 go test -run "TestAccPagerDutyServiceEventRule" ./pagerduty -v -timeout 120m
=== RUN   TestAccPagerDutyServiceEventRule_import
--- PASS: TestAccPagerDutyServiceEventRule_import (10.92s)
=== RUN   TestAccPagerDutyServiceEventRule_Basic
--- PASS: TestAccPagerDutyServiceEventRule_Basic (13.19s)
=== RUN   TestAccPagerDutyServiceEventRule_MultipleRules
--- PASS: TestAccPagerDutyServiceEventRule_MultipleRules (14.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-pagerduty/pagerduty	39.888s
```

And for `pagerduty_ruleset_rule` here:

```
TF_ACC=1 go test -run "TestAccPagerDutyRulesetRule" ./pagerduty -v -timeout 120m     
=== RUN   TestAccPagerDutyRulesetRule_import
--- PASS: TestAccPagerDutyRulesetRule_import (5.23s)
=== RUN   TestAccPagerDutyRulesetRule_Basic
--- PASS: TestAccPagerDutyRulesetRule_Basic (5.90s)
=== RUN   TestAccPagerDutyRulesetRule_MultipleRules
--- PASS: TestAccPagerDutyRulesetRule_MultipleRules (6.43s)
PASS
ok  	github.com/terraform-providers/terraform-provider-pagerduty/pagerduty	18.212s
```

### CODE REVIEW

File: vendor/github.com/heimweh/go-pagerduty/pagerduty/pagerduty.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type Client struct {
 	ServiceDependencies *ServiceDependencyService
 	Priorities          *PriorityService
 	ResponsePlays       *ResponsePlayService
+	Tags                *TagService
```
Comment (dobs) [inline vendor/github.com/heimweh/go-pagerduty/pagerduty/pagerduty.go:55]: Gut check -- is adding tags intentional? Not mentioned in the PR description.

**Review by dobs — APPROVED** [review]
LGTM -- only thing to confirm is whether the tags stuff is intended to be included in this PR as well. Otherwise just two nits.

File: website/docs/r/ruleset_rule.html.markdown [hunk 2]
```diff
@@ -106,7 +107,8 @@ The following arguments are supported:
 	* `threshold_value` - The number of alerts that should be suppressed.
 	* `threshold_time_amount` - The number value of the `threshold_time_unit` before an incident is created.
 	* `threshold_time_unit` - The `minutes`,`hours`, or `days` that the `threshold_time_amount` should be measured. 
-
+* `event_action` (Optional) - An object with a single `value` field. The value sets whether the resulting alert status is `trigger` or `resolve`.
+* `suspend` (Optional) - An object with a single `value` field. The value sets the length of time to suspend the resulting alert before triggering.
```
[the remark this answers is not in the document]
Comment (stmcallister) [reply website/docs/r/ruleset_rule.html.markdown]: Good find. updated to add whitespace here.

File: pagerduty/resource_pagerduty_ruleset_rule_test.go [hunk 3]
```diff
@@ -31,6 +31,8 @@ func TestAccPagerDutyRulesetRule_Basic(t *testing.T) {
 						"pagerduty_ruleset_rule.foo", "disabled", "true"),
 					resource.TestCheckResourceAttr(
 						"pagerduty_ruleset_rule.foo", "conditions.#", "1"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_ruleset_rule.foo", "variable.#", "2"),
```
[the remark this answers is not in the document]
Comment (stmcallister) [reply pagerduty/resource_pagerduty_ruleset_rule_test.go:33]: I agree with that thinking. Updated to group the checks a little better.

File: vendor/github.com/heimweh/go-pagerduty/pagerduty/pagerduty.go (hunk 1 above)
Comment (stmcallister) [reply vendor/github.com/heimweh/go-pagerduty/pagerduty/pagerduty.go]: removed Tags from client for now.

**Pull request merged**